logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendTextMessageUseCase.kt

```diff
@@ -11,31 +12,37 @@ import com.wire.kalium.logic.functional.Either
 import com.wire.kalium.logic.functional.suspending
 import com.wire.kalium.logic.sync.SyncManager
 import kotlinx.coroutines.flow.first
+import kotlinx.datetime.Clock
 
 class SendTextMessageUseCase(
     private val messageRepository: MessageRepository,
     private val userRepository: UserRepository,
     private val clientRepository: ClientRepository,
-    private val syncManager: SyncManager
+    private val syncManager: SyncManager,
+    private val messageSender: MessageSender
 ) {
 
     suspend operator fun invoke(conversationId: ConversationId, text: String): Either<CoreFailure, Unit> {
         syncManager.waitForSlowSyncToComplete()
         val selfUser = userRepository.getSelfUser().first()
 
         return suspending {
-            clientRepository.currentClientId().flatMap { currentClientId ->
+            val generatedMessageUuid = uuid4().toString()
 
+            clientRepository.currentClientId().flatMap { currentClientId ->
                 val message = Message(
-                    id = "someUUID",
+                    id = generatedMessageUuid,
                     content = MessageContent.Text(text),
                     conversationId = conversationId,
-                    date = "25 Jan 2022 13:30:00 GMT",
+                    // TODO: Do we care about clock sync? Should we get server time?
+                    date = Clock.System.now().toString(),
```

Comment:
AFAIK this date is only for our own internal ordering of messages. On iOS we don't care about clock sync except when dealing with calling messages but that happens inside AVS we just provide the server time.